astropy/coordinates/angles/_wrap.pyx

```diff
@@ -0,0 +1,55 @@
+#cython: language_level=3
+cimport cython
+cimport numpy as np
+from libc cimport math
+
+np.import_array()
+
+ctypedef fused dtype:
+    short
+    int
+    long
+    float
+    double
+
+
+@cython.ufunc
+@cython.cdivision(True)
+cdef dtype wrap_at(dtype angle, dtype wrap_angle, dtype period):
+    cdef dtype wrap_angle_floor
+    wrap_angle_floor = wrap_angle - period
+
+    if dtype is float or dtype is double:
+        if math.isnan(angle):
+            return angle
+
+    if (angle >= wrap_angle_floor) and (angle < wrap_angle):
+        return angle
+
+    cdef int wraps
```

Comment:
is `int` guaranteed to be 4 bytes long in Cython ? I'm worried that using platform-dependent-sized types might lead to very subtle errors in the distant future.

Reply:
It is the native c type, but I really don't expect issues here. It would mean that a value wraps more often than 2 * pi * int max, which seems impossible...

Reply:
good point. I've been bitten by this kind of problem before which is why I may have jumped on this one a bit quickly. Agreed that it's unlikely enough that it doesn't matter in practice !